sway/desktop/output.c

```diff
@@ -745,12 +745,18 @@ static void damage_child_views_iterator(struct sway_container *con,
 
 void output_damage_whole_container(struct sway_output *output,
 		struct sway_container *con) {
+	int shadow_sigma = con->shadow_enabled ? config->shadow_blur_sigma : 0;
+	int blur_size = con->blur_enabled ?
+			pow(2, config->blur_params.num_passes) * config->blur_params.radius : 0;
+	// +1 as a margin of error
+	int effect_size = MAX(shadow_sigma, blur_size) + 1;
+
 	// Pad the box by 1px, because the width is a double and might be a fraction
 	struct wlr_box box = {
-		.x = con->current.x - output->lx - 1,
-		.y = con->current.y - output->ly - 1,
-		.width = con->current.width + 2,
-		.height = con->current.height + 2,
+		.x = con->current.x - output->lx - 1 - effect_size,
+		.y = con->current.y - output->ly - 1 - effect_size,
+		.width = con->current.width + 2 + effect_size * 2,
+		.height = con->current.height + 2 + effect_size * 2,
```

Comment:
We shouldn't expand the damage all the time. We should only expand the region if there's a window that can take advantage of the blur

Reply:
Yeah, should add a few extra checks in addition to if the container has blur enabled